rust/lance/src/io/exec/planner.rs

```diff
@@ -517,11 +517,7 @@ impl Planner {
     /// Note: the returned expression must be passed through [optimize_expr()]
     /// before being passed to [create_physical_expr()].
     pub fn parse_filter(&self, filter: &str) -> Result<Expr> {
-        // Allow sqlparser to parse filter as part of ONE SQL statement.
-        let ast_expr = parse_sql_filter(filter)?;
-        let expr = self.parse_sql_expr(&ast_expr)?;
-        let schema = Schema::try_from(self.schema.as_ref())?;
-        let resolved = resolve_expr(&expr, &schema)?;
+        let resolved = self.parse_expr(filter)?;
```

Comment:
If your changes don't rely on this, I'd rather keep using `parse_sql_filter` as is. We can consider eliminating that function in a separate PR.